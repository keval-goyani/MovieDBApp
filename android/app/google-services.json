{
  "project_info": {
    "project_number": "89996564819",
    "project_id": "moviedb-2908d",
    "storage_bucket": "moviedb-2908d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:89996564819:android:e942630a87714b1d3ea9a6",
        "android_client_info": {
          "package_name": "com.moviedbapp"
        }
      },
      "oauth_client": [
        {
          "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBYrM1MC9Go35l8ra748zqBtCRDkrB_Nn4"
        }
      ],
      "services": {
        "appinvite_service": {
          "other_platform_oauth_client": [
            {
              "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
              "client_type": 3
            }
          ]
        }
      }
    }
  ],
  "configuration_version": "1"
}