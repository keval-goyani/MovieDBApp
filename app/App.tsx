import React, {useEffect} from 'react';
import {SafeAreaView, Text} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {styles} from './theme';

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Text>Hello World</Text>
    </SafeAreaView>
  );
};

export default App;
