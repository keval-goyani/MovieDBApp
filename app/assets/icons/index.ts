export default {
  menuIcon: require('./menuIcon.png'),
  searchIcon: require('./searchIcon.png'),
  movieDbIcon: require('./movieDBIcon.png'),
  expandIcon: require('./expandIcon.png'),
  threeDotIcon: require('./threeDotIcon.png'),
  playIcon: require('./playIcon.png'),
  backIcon: require('./backIcon.png'),
  movieDbIcon3x: require('./movieDB3x.png'),
  movieDbIcon2x: require('./movieDB2x.png'),
  homeIcon: require('./homeIcon.png'),
  trailerIcon: require('./trailerIcon.png'),
  moviesIcon: require('./moviesIcon.png'),
  communityIcon: require('./communityIcon.png'),
  notFound: require('./notFoundIcon.png'),
  comingSoon: require('./comingSoon.png'),
  logout: require('./logout.png'),
  trending: require('./trending.png'),
  tv: require('./tv.png'),
  avatar: require('./user.png'),
  chatBackground: require('./chatBackground.png'),
  send: require('./send.png'),
  location: require('./share.png'),
  camera: require('./camera.png'),
  attach: require('./attach.png'),
  gallery: require('./gallery.png'),
  dotMenu: require('./dots.png'),
};
