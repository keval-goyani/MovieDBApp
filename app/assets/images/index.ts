export default {
  docFirstPage: require('./docFirstPage.png'),
};
