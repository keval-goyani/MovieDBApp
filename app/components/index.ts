import ChatHeader from './ChatHeader';
import ChatInput from './ChatInput';
import CustomSkeleton from './CustomSkeleton';
import DropDownMenu from './DropDownMenu';
import Header from './Header';
import List from './List';
import ListContainer, { movieListItem } from './ListContainer';
import Loader from './Loader';
import LoadingState from './LoadingState';
import Message from './Message';
import MessageList from './MessageList';
import Movies from './Movies';
import MovieTrailer, { trailerListItem } from './MovieTrailers';
import SearchModal from './SearchModal';
import SkeletonCard from './SkeletonCard';
import TabBarIcon from './TabBarIcon';
import Trailers from './Trailers';
import UserListEmpty from './UserListEmpty';
import UsersList from './UsersList';

export {
  ChatHeader,
  ChatInput,
  DropDownMenu,
  Header,
  List,
  ListContainer,
  MovieTrailer,
  Message,
  MessageList,
  TabBarIcon,
  SearchModal,
  Loader,
  LoadingState,
  Movies,
  movieListItem,
  Trailers,
  trailerListItem,
  UsersList,
  SkeletonCard,
  CustomSkeleton,
  UserListEmpty,
};
