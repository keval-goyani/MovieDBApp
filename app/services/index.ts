import { getPopularMovieData } from './Api';
import immutablePersistenceTransform from './immutablePersistenceTransform';
import {
  alertMessage,
  apiConfig,
  loginError,
  chatCreation,
  clearChat,
  conversationIdCreation,
  convertToTimestamp,
  decryptData,
  encryptData,
  getChatTime,
  getDetails,
  getError,
  getUniqueMovies,
  handleCameraPermission,
  handleDocumentPermission,
  handleGalleryPermission,
  openDocument,
  searchMovie,
  signUpError,
  sortString,
  timestampToTime,
  alertBox,
  storeImageToStorage,
  groupCreation,
  dateConvertor,
  clearConversation,
  sub,
  handleLogOut,
  chatImageHandler,
} from './Utils';

export {
  immutablePersistenceTransform,
  alertMessage,
  apiConfig,
  getDetails,
  getError,
  getPopularMovieData,
  searchMovie,
  getUniqueMovies,
  getChatTime,
  timestampToTime,
  sortString,
  handleGalleryPermission,
  handleCameraPermission,
  clearChat,
  chatCreation,
  handleDocumentPermission,
  conversationIdCreation,
  openDocument,
  encryptData,
  decryptData,
  convertToTimestamp,
  loginError,
  signUpError,
  alertBox,
  storeImageToStorage,
  groupCreation,
  dateConvertor,
  clearConversation,
  sub,
  handleLogOut,
  chatImageHandler,
};
