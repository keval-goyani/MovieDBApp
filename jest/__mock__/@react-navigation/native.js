export const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  replace: jest.fn(),
  popToTop: jest.fn(),
  addListener: jest.fn().mockImplementation((str, cb) => cb()),
  pop: jest.fn(),
  getParam: jest.fn(),
  push: jest.fn(),
};

const useRoute = jest.fn(val => val);
// const useFocusEffect = jest.fn();
const useIsFocused = jest.fn(val => val);
const useNavigation = () => navigation;
const setFocus = jest.fn();
const createNavigationContainerRef = jest.fn();
const DefaultTheme = {};

const CommonActions = { reset: jest.fn() };
const StackActions = {
  pop: jest.fn(),
};

export {
  useRoute,
  useNavigation,
  // useFocusEffect,
  useIsFocused,
  setFocus,
  createNavigationContainerRef,
  DefaultTheme,
  CommonActions,
  StackActions,
};
