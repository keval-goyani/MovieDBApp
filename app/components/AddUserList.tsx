import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useEffect, useState } from 'react';
import { FlatList, Text, TouchableOpacity, View } from 'react-native';
import Check from 'react-native-vector-icons/AntDesign';
import { useDispatch, useSelector } from 'react-redux';
import { asMutable } from 'seamless-immutable';
import { ProfileImage, UserListEmpty } from '../components';
import {
  AddUserListProps,
  NavigationDataType,
  navigationStrings,
  RenderItemTypes,
  SelectedUsersProps,
  strings,
  UserDataType,
} from '../constants';
import { authDataSelectors } from '../redux/AuthRedux';
import { userListSelector } from '../redux/UserListRedux';
import { conversationIdCreation, sub } from '../services';
import { Color, moderateScale } from '../theme';
import { styles } from './styles/AddUserListStyles';

const AddUsersList = ({ userListData, setSelectedUsers }: AddUserListProps) => {
  const navigation: NavigationDataType = useNavigation();
  const dispatch = useDispatch();
  const { user } = useSelector(authDataSelectors.getData);
  const { fetchingUserList } = useSelector(userListSelector.getData);
  const [selectedItems, setSelectedItems] = useState<UserDataType[]>([]);

  useFocusEffect(() => {
    sub({ user, dispatch });
    return () => sub;
  });

  useEffect(() => {
    setSelectedUsers(selectedItems);
  }, [selectedItems, setSelectedUsers]);

  const renderUserList = ({ item, index }: RenderItemTypes) => {
    const usersEmail = [user?.email ?? '', item?.email];
    const conversationId: string = conversationIdCreation(usersEmail);

    const handleOnPress = (selectedUser: UserDataType) => {
      console.log(selectedItems, '<==selectedItems.length');
      console.log(selectedUser, '<==selectedUser');

      if (selectedItems.length) {
        return selectUsers(selectedUser);
      }
      navigation.goBack();
      navigation.navigate(navigationStrings.Chat, {
        conversationId,
        username: item?.username,
        receiverId: item?.uid,
        userStatus: item?.status,
        profileImage: item?.profileImage,
        userEmail: item?.email,
      });
    };

    const getSelected = (usersData: SelectedUsersProps) => {
      return JSON.stringify(selectedItems).includes(JSON.stringify(usersData));
    };

    const selected = getSelected(item);

    const selectUsers = (selectedUser: UserDataType) => {
      if (
        JSON.stringify(selectedItems).includes(JSON.stringify(selectedUser))
      ) {
        const filteredList = selectedItems?.filter(
          (items: UserDataType) => items !== selectedUser,
        );
        setSelectedItems(filteredList);
      } else {
        setSelectedItems([...selectedItems, selectedUser]);
      }
    };

    return (
      <TouchableOpacity
        style={styles.listItem}
        onPress={() => handleOnPress(item)}x
        activeOpacity={0.8}
        onLongPress={() => selectUsers(item)}
        testID={`button-for-test${index}`}>
        <View style={styles.itemContainer}>
          <ProfileImage
            profileImage={item?.profileImage}
            userStatus={item?.status}
          />
          <View style={styles.nameView}>
            <Text style={styles.text}>{item?.username}</Text>
            <Text
              style={styles.lastChatText}
              ellipsizeMode="tail"
              numberOfLines={1}>
              {item?.email}
            </Text>
          </View>
        </View>
        {selected && (
          <>
            <View style={styles.overlay} />
            <Check
              testID="check"
              name={strings.checkIcon}
              size={moderateScale(23)}
              color={Color.lightBlue}
              style={styles.iconStyle}
            />
          </>
        )}
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      testID={'flatList'}
      data={asMutable(userListData)}
      renderItem={renderUserList}
      ListEmptyComponent={
        <UserListEmpty
          fetching={fetchingUserList}
          userListLength={userListData.length}
        />
      }
      showsVerticalScrollIndicator={false}
      bounces={false}
    />
  );
};

export default AddUsersList;
