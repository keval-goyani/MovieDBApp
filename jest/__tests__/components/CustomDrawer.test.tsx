import { DrawerItem } from '@react-navigation/drawer';
import { useNavigation } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import React, { useState } from 'react';
import { Alert } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { Icons } from '../../../app/assets';
import { CustomDrawer } from '../../../app/components';
import selectedAction from '../../../app/redux/DrawerSelectRedux';
import reduxStore from '../../../app/redux/store';
import { handleLogOut } from '../../../app/services';

describe('Custom Drawer', () => {
  const setActiveTabMock = { payload: 'Home', type: 'SELECTED' };
  beforeEach(() => {
    useState.mockImplementation(jest.requireActual('react').useState);
    useSelector.mockReturnValue({
      setActiveTab: setActiveTabMock,
    });
    const mockDispatch = jest.fn();
    useDispatch.mockReturnValue(mockDispatch);
  });

  const navigation = useNavigation();
  it('should take snapshot of Custom Drawer', () => {
    const { toJSON } = render(<CustomDrawer navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render profileImage when user have profileImage set', () => {
    const setActiveTabMocks = { payload: 'Home', type: 'SELECTED' };
    const userData = { profileImage: 'http://image.com' };

    useSelector.mockReturnValue({
      setActiveTab: setActiveTabMocks,
      user: userData,
    });
    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);

    expect(getByTestId('user-image')).toBeDefined();
  });

  it('should render defaultProfile image when user has not set the profileImage', () => {
    const setActiveTabMocks = { payload: 'Home', type: 'SELECTED' };
    const userData = { username: 'chandnai' };

    useSelector.mockReturnValue({
      setActiveTab: setActiveTabMocks,
      user: userData,
    });
    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);
    expect(getByTestId('default-profile-image')).toBeDefined();
  });

  it('should render the edit profile button', () => {
    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);
    expect(getByTestId('edit-profile')).toBeDefined();
  });

  it('should open EditProfile modal on clicking Edit Profile button', () => {
    const setIsOpen = jest.fn();
    const setImagePath = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy
      .mockImplementationOnce(() => [true, setIsOpen])
      .mockImplementationOnce(() => [true, setIsOpen])
      .mockImplementationOnce(() => ['http://image.com', setImagePath]);


    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);
    const EditProfileButton = getByTestId('edit-profile');
    fireEvent.press(EditProfileButton);
    expect(getByTestId('edit-profile-modal')).toBeDefined();
  });

  it('should navigate to home screen on clicking to Home drawer item and icon should be There', () => {
    const dispatch = useDispatch();
    const { container, debug } = render(
      <CustomDrawer navigation={navigation} />,
    );
    debug();
    console.log(debug(), '<==debug');

    const drawerItem = container.findAllByType(DrawerItem);
    const homeDrawerItem = drawerItem.find(item => item.props.label === 'Home');
    fireEvent.press(homeDrawerItem);
    expect(dispatch).toHaveBeenCalledWith(selectedAction.selected('Home'));
    expect(navigation.navigate).toHaveBeenCalledWith('TabRoot', {
      screen: 'Home',
    });

    const homeIcon = drawerItem
      .find(item => item.props.label === 'Home')
      ?.props.icon();

    expect(homeIcon.props.source).toBeTruthy();
    expect(homeIcon.props.source.testUri).toBe(Icons.homeIcon.testUri);
  });

  it('should navigate to Trailers screen on clicking to Trending drawer item icon should be There', () => {
    const dispatch = useDispatch();
    const { container } = render(<CustomDrawer navigation={navigation} />);

    const drawerItem = container.findAllByType(DrawerItem);
    const trendingDrawerItem = drawerItem.find(
      item => item.props.label === 'Trending',
    );
    fireEvent.press(trendingDrawerItem);
    expect(dispatch).toHaveBeenCalledWith(selectedAction.selected('Trailers'));
    expect(navigation.navigate).toHaveBeenCalledWith('TabRoot', {
      screen: 'Trailers',
    });

    const trendingIcon = drawerItem
      .find(item => item.props.label === 'Trending')
      ?.props.icon();

    expect(trendingIcon.props.source).toBeTruthy();
    expect(trendingIcon.props.source.testUri).toEqual(Icons.trending.testUri);
  });

  it('should navigate to Movies screen on clicking to Movies drawer item and Icon should be there', () => {
    const dispatch = useDispatch();
    const { container } = render(<CustomDrawer navigation={navigation} />);

    const drawerItem = container.findAllByType(DrawerItem);
    const moviesDrawerItem = drawerItem.find(
      item => item.props.label === 'Movies',
    );
    fireEvent.press(moviesDrawerItem);
    expect(dispatch).toHaveBeenCalledWith(selectedAction.selected('Movies'));
    expect(navigation.navigate).toHaveBeenCalledWith('TabRoot', {
      screen: 'Movies',
    });

    const moviesIcon = drawerItem
      .find(item => item.props.label === 'Movies')
      ?.props.icon();

    expect(moviesIcon.props.source).toBeTruthy();
    expect(moviesIcon.props.source.testUri).toEqual(Icons.moviesIcon.testUri);
  });

  it('should just call onPress Function clicking to TV Shows drawer item and Icon should be there', () => {
    const { container } = render(<CustomDrawer navigation={navigation} />);

    const drawerItem = container.findAllByType(DrawerItem);
    const tvShowsDrawerItem = drawerItem.find(
      item => item.props.label === 'TV Shows',
    );
    fireEvent.press(tvShowsDrawerItem);

    const tvShowsIcon = drawerItem
      .find(item => item.props.label === 'TV Shows')
      ?.props.icon();

    expect(tvShowsIcon.props.source).toBeTruthy();
    expect(tvShowsIcon.props.source.testUri).toEqual(Icons.tv.testUri);
  });

  it('should navigate to Community screen on clicking to People drawer item', () => {
    const dispatch = useDispatch();
    const { container } = render(<CustomDrawer navigation={navigation} />);

    const drawerItem = container.findAllByType(DrawerItem);
    const peopleDrawerItem = drawerItem.find(
      item => item.props.label === 'People',
    );
    fireEvent.press(peopleDrawerItem);
    expect(dispatch).toHaveBeenCalledWith(selectedAction.selected('Community'));
    expect(navigation.navigate).toHaveBeenCalledWith('TabRoot', {
      screen: 'Community',
    });

    const communityIcon = drawerItem
      .find(item => item.props.label === 'People')
      ?.props.icon();

    expect(communityIcon.props.source).toBeTruthy();
    expect(communityIcon.props.source.uri).toBe(Icons?.communityIcon.uri);
  });

  it('should popUp Alert on pressing the Logout button', () => {
    const dispatch = useDispatch();
    const { persistor } = reduxStore;

    const spy = jest.spyOn(Alert, 'alert').mockImplementation();
    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);
    const logoutBUtton = getByTestId('logout-button');
    fireEvent.press(logoutBUtton);

    expect(spy).toHaveBeenCalledWith('Warning', 'Confirm Logout?', [
      {
        text: 'Cancel',
        style: 'cancel',
      },
      {
        text: 'OK',
        onPress: handleLogOut({ dispatch, persistor }),
      },
    ]);
  });

  it('should call onPressIn with selected style', () => {
    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);
    const logoutBUtton = getByTestId('logout-button');

    fireEvent(logoutBUtton, 'onPressIn');
    expect(logoutBUtton.props.style.backgroundColor).toEqual('#32d9cb');
  });

  it('should call onPressOut with selected style', () => {
    const { getByTestId } = render(<CustomDrawer navigation={navigation} />);
    const logoutBUtton = getByTestId('logout-button');

    fireEvent(logoutBUtton, 'onPressOut');
    expect(logoutBUtton.props.style.backgroundColor).toEqual('#0d253f');
  });
});
