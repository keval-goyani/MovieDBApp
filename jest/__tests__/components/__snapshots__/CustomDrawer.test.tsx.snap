// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Drawer should take snapshot of Custom Drawer 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#0d253f",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": undefined,
          "borderColor": "#FFFFFF",
          "flex": 1,
          "justifyContent": "center",
          "marginBottom": undefined,
          "paddingVertical": undefined,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#01b4e4",
              "borderRadius": 50,
              "borderWidth": undefined,
              "height": undefined,
              "justifyContent": "center",
              "width": undefined,
            }
          }
        >
          <Image
            source={
              Object {
                "default": Object {
                  "attach": Object {
                    "testUri": "../../../app/assets/icons/attach.png",
                  },
                  "avatar": Object {
                    "testUri": "../../../app/assets/icons/user.png",
                  },
                  "backIcon": Object {
                    "testUri": "../../../app/assets/icons/backIcon.png",
                  },
                  "camera": Object {
                    "testUri": "../../../app/assets/icons/camera.png",
                  },
                  "cameraIcon": Object {
                    "testUri": "../../../app/assets/icons/cameraIcon.png",
                  },
                  "chatBackground": Object {
                    "testUri": "../../../app/assets/icons/chatBackground.png",
                  },
                  "close": Object {
                    "testUri": "../../../app/assets/icons/close.png",
                  },
                  "comingSoon": Object {
                    "testUri": "../../../app/assets/icons/comingSoon.png",
                  },
                  "communityIcon": Object {
                    "testUri": "../../../app/assets/icons/communityIcon.png",
                  },
                  "document": Object {
                    "testUri": "../../../app/assets/icons/documentIcon.png",
                  },
                  "documentIcon": Object {
                    "testUri": "../../../app/assets/icons/documentIcon.png",
                  },
                  "dotMenu": Object {
                    "testUri": "../../../app/assets/icons/dots.png",
                  },
                  "expandIcon": Object {
                    "testUri": "../../../app/assets/icons/expandIcon.png",
                  },
                  "fileIcon": Object {
                    "testUri": "../../../app/assets/icons/fileIcon.png",
                  },
                  "fullStop": Object {
                    "testUri": "../../../app/assets/icons/fullStop.png",
                  },
                  "galleryIcon": Object {
                    "testUri": "../../../app/assets/icons/galleryIcon.png",
                  },
                  "homeIcon": Object {
                    "testUri": "../../../app/assets/icons/homeIcon.png",
                  },
                  "imageDocument": Object {
                    "testUri": "../../../app/assets/icons/imageIcon.png",
                  },
                  "location": Object {
                    "testUri": "../../../app/assets/icons/share.png",
                  },
                  "locationIcon": Object {
                    "testUri": "../../../app/assets/icons/locationIcon.png",
                  },
                  "logout": Object {
                    "testUri": "../../../app/assets/icons/logout.png",
                  },
                  "menuIcon": Object {
                    "testUri": "../../../app/assets/icons/menuIcon.png",
                  },
                  "movieDbIcon": Object {
                    "testUri": "../../../app/assets/icons/movieDBIcon.png",
                  },
                  "movieDbIcon2x": Object {
                    "testUri": "../../../app/assets/icons/movieDB2x.png",
                  },
                  "movieDbIcon3x": Object {
                    "testUri": "../../../app/assets/icons/movieDB3x.png",
                  },
                  "moviesIcon": Object {
                    "testUri": "../../../app/assets/icons/moviesIcon.png",
                  },
                  "noItemFound": Object {
                    "testUri": "../../../app/assets/icons/noItemFound.png",
                  },
                  "notFound": Object {
                    "testUri": "../../../app/assets/icons/notFoundIcon.png",
                  },
                  "pdfDocument": Object {
                    "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                  },
                  "playIcon": Object {
                    "testUri": "../../../app/assets/icons/playIcon.png",
                  },
                  "plus": Object {
                    "testUri": "../../../app/assets/icons/plusSign.png",
                  },
                  "pptDocument": Object {
                    "testUri": "../../../app/assets/icons/pptDocument.png",
                  },
                  "searchIcon": Object {
                    "testUri": "../../../app/assets/icons/searchIcon.png",
                  },
                  "send": Object {
                    "testUri": "../../../app/assets/icons/send.png",
                  },
                  "threeDotIcon": Object {
                    "testUri": "../../../app/assets/icons/threeDotIcon.png",
                  },
                  "trailerIcon": Object {
                    "testUri": "../../../app/assets/icons/trailerIcon.png",
                  },
                  "trending": Object {
                    "testUri": "../../../app/assets/icons/trending.png",
                  },
                  "tv": Object {
                    "testUri": "../../../app/assets/icons/tv.png",
                  },
                  "videoDocument": Object {
                    "testUri": "../../../app/assets/icons/mp4Document.png",
                  },
                },
              }
            }
            style={
              Object {
                "height": undefined,
                "tintColor": "#01b4e4",
                "width": undefined,
              }
            }
            testID="default-profile-image"
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#32d9cb",
              "borderRadius": 20,
              "bottom": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
            }
          }
          testID="edit-profile"
        >
          <Icon
            color="#0d253f"
            name="account-edit-outline"
            style={
              Object {
                "margin": undefined,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#32d9cb",
            "fontSize": undefined,
          }
        }
      />
    </View>
    <View />
    <View />
    <View />
    <View />
    <View />
    <View
      style={
        Object {
          "marginTop": undefined,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0d253f",
            "borderRadius": 4,
            "marginHorizontal": undefined,
            "opacity": 1,
            "paddingVertical": undefined,
          }
        }
        testID="logout-button"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "default": Object {
                  "attach": Object {
                    "testUri": "../../../app/assets/icons/attach.png",
                  },
                  "avatar": Object {
                    "testUri": "../../../app/assets/icons/user.png",
                  },
                  "backIcon": Object {
                    "testUri": "../../../app/assets/icons/backIcon.png",
                  },
                  "camera": Object {
                    "testUri": "../../../app/assets/icons/camera.png",
                  },
                  "cameraIcon": Object {
                    "testUri": "../../../app/assets/icons/cameraIcon.png",
                  },
                  "chatBackground": Object {
                    "testUri": "../../../app/assets/icons/chatBackground.png",
                  },
                  "close": Object {
                    "testUri": "../../../app/assets/icons/close.png",
                  },
                  "comingSoon": Object {
                    "testUri": "../../../app/assets/icons/comingSoon.png",
                  },
                  "communityIcon": Object {
                    "testUri": "../../../app/assets/icons/communityIcon.png",
                  },
                  "document": Object {
                    "testUri": "../../../app/assets/icons/documentIcon.png",
                  },
                  "documentIcon": Object {
                    "testUri": "../../../app/assets/icons/documentIcon.png",
                  },
                  "dotMenu": Object {
                    "testUri": "../../../app/assets/icons/dots.png",
                  },
                  "expandIcon": Object {
                    "testUri": "../../../app/assets/icons/expandIcon.png",
                  },
                  "fileIcon": Object {
                    "testUri": "../../../app/assets/icons/fileIcon.png",
                  },
                  "fullStop": Object {
                    "testUri": "../../../app/assets/icons/fullStop.png",
                  },
                  "galleryIcon": Object {
                    "testUri": "../../../app/assets/icons/galleryIcon.png",
                  },
                  "homeIcon": Object {
                    "testUri": "../../../app/assets/icons/homeIcon.png",
                  },
                  "imageDocument": Object {
                    "testUri": "../../../app/assets/icons/imageIcon.png",
                  },
                  "location": Object {
                    "testUri": "../../../app/assets/icons/share.png",
                  },
                  "locationIcon": Object {
                    "testUri": "../../../app/assets/icons/locationIcon.png",
                  },
                  "logout": Object {
                    "testUri": "../../../app/assets/icons/logout.png",
                  },
                  "menuIcon": Object {
                    "testUri": "../../../app/assets/icons/menuIcon.png",
                  },
                  "movieDbIcon": Object {
                    "testUri": "../../../app/assets/icons/movieDBIcon.png",
                  },
                  "movieDbIcon2x": Object {
                    "testUri": "../../../app/assets/icons/movieDB2x.png",
                  },
                  "movieDbIcon3x": Object {
                    "testUri": "../../../app/assets/icons/movieDB3x.png",
                  },
                  "moviesIcon": Object {
                    "testUri": "../../../app/assets/icons/moviesIcon.png",
                  },
                  "noItemFound": Object {
                    "testUri": "../../../app/assets/icons/noItemFound.png",
                  },
                  "notFound": Object {
                    "testUri": "../../../app/assets/icons/notFoundIcon.png",
                  },
                  "pdfDocument": Object {
                    "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                  },
                  "playIcon": Object {
                    "testUri": "../../../app/assets/icons/playIcon.png",
                  },
                  "plus": Object {
                    "testUri": "../../../app/assets/icons/plusSign.png",
                  },
                  "pptDocument": Object {
                    "testUri": "../../../app/assets/icons/pptDocument.png",
                  },
                  "searchIcon": Object {
                    "testUri": "../../../app/assets/icons/searchIcon.png",
                  },
                  "send": Object {
                    "testUri": "../../../app/assets/icons/send.png",
                  },
                  "threeDotIcon": Object {
                    "testUri": "../../../app/assets/icons/threeDotIcon.png",
                  },
                  "trailerIcon": Object {
                    "testUri": "../../../app/assets/icons/trailerIcon.png",
                  },
                  "trending": Object {
                    "testUri": "../../../app/assets/icons/trending.png",
                  },
                  "tv": Object {
                    "testUri": "../../../app/assets/icons/tv.png",
                  },
                  "videoDocument": Object {
                    "testUri": "../../../app/assets/icons/mp4Document.png",
                  },
                },
              }
            }
            style={
              Array [
                Object {
                  "height": undefined,
                  "marginLeft": undefined,
                  "resizeMode": "contain",
                  "tintColor": "#00bfff",
                  "width": undefined,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#00bfff",
                "fontSize": undefined,
                "fontWeight": "500",
                "marginLeft": undefined,
              }
            }
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
